### PR TITLE
Partially Translated protocols.asc

### PR DESCRIPTION
**Pull request by jjrang11**
Partially translated 1,254 words for progit2/book/04-git-server/sections/protocols.asc

### CODE REVIEW

File: book/04-git-server/sections/protocols.asc [hunk 1]
```diff
@@ -1,100 +1,100 @@
-=== The Protocols
+=== Ang Mga Protokol
+Maaaring gamitin ng Git ang apat na magkakaibang mga protokol upang lumipat ng mga data: Lokal, HTTP, Secure Shell (SSH) at Git.
+Dito tatalakayin natin kung ano sila at kung saang mga pangunahing pangyayari mo gusto (o di gusto) sila gamitin.
```
Comment (mind-blast) [inline book/04-git-server/sections/protocols.asc]: mo gusto to gusto mo
sila gamitin to silang gamitin

File: book/04-git-server/sections/protocols.asc [hunk 2]
```diff
@@ -1,100 +1,100 @@
...
+Kapag ikaw ay may isang ibinahagi na naka-mount na filesystem, maaari kang mag clone, i-push sa, at mag pull mula sa isang lokal na nakabatay sa file na repositoryo.
+Para i-clone ang isang repositoryo na tulad nito, o idagdag bilang isang remote sa umiiral na proyekto, gamitin ang landas patungo sa repositoryo bilang URL.
+Halimbawa, upang i-clone ang isang lokal na repositoryo, maaari kang magpagana ng ganito:
 
-If you have a shared mounted filesystem, then you can clone, push to, and pull from a local file-based repository.
-To clone a repository like this, or to add one as a remote to an existing project, use the path to the repository as the URL.
-For example, to clone a local repository, you can run something like this:
-
-[source,console]
+[pinagmulan,console]
```
Comment (mind-blast) [inline book/04-git-server/sections/protocols.asc]: [pinagmulan,console] <-- no need to change this leave it as [source,console]

File: book/04-git-server/sections/protocols.asc [hunk 3]
```diff
@@ -1,100 +1,100 @@
...
 ----
 $ git clone /srv/git/project.git
 ----
 
-Or you can do this:
+O maaari mong gawin ito:
 
-[source,console]
+[pinagmulan,console]
```
Comment (mind-blast) [inline book/04-git-server/sections/protocols.asc]: [pinagmulan,console] to [source,console]

File: book/04-git-server/sections/protocols.asc [hunk 4]
```diff
@@ -1,100 +1,100 @@
...
-Git can communicate over HTTP using two different modes.
-Prior to Git 1.6.6, there was only one way it could do this which was very simple and generally read-only.
-In version 1.6.6, a new, smarter protocol was introduced that involved Git being able to intelligently negotiate data transfer in a manner similar to how it does over SSH.
-In the last few years, this new HTTP protocol has become very popular since it's simpler for the user and smarter about how it communicates.
-The newer version is often referred to as the _Smart_ HTTP protocol and the older way as _Dumb_ HTTP.
-We'll cover the newer Smart HTTP protocol first.
+Ang Git ay maaaring makipag-usap sa HTTP gamit ang dalawang makibang mga mode.
+Bago ang Git 1.6.6, Mayroon lamang isang paraan upang maisagawa ito na kung saan ay madali at sa pangkalahatan ay read-only.
+Sa bersyon 1.6.6, isang bago, mas matalinong protokol ang ipinakilala na kasangkot ang Git na magagawang mas matalino na pag-usap ng pag lipat ng data sa parehong paraan ng SSH. 
+Sa mga nakaraang mga taon, itong bago na protokol ng HTTP ay naging popular dahil ito ay mas madali para sa user at mas matalino kung papaano nakikipag-usap.
```
Comment (mind-blast) [inline book/04-git-server/sections/protocols.asc]: user to gumagamit

File: book/04-git-server/sections/protocols.asc [hunk 5]
```diff
@@ -1,100 +1,100 @@
...
-==== The HTTP Protocols
+==== Ang mga Protokol ng HTTP
 
-Git can communicate over HTTP using two different modes.
-Prior to Git 1.6.6, there was only one way it could do this which was very simple and generally read-only.
-In version 1.6.6, a new, smarter protocol was introduced that involved Git being able to intelligently negotiate data transfer in a manner similar to how it does over SSH.
-In the last few years, this new HTTP protocol has become very popular since it's simpler for the user and smarter about how it communicates.
-The newer version is often referred to as the _Smart_ HTTP protocol and the older way as _Dumb_ HTTP.
-We'll cover the newer Smart HTTP protocol first.
+Ang Git ay maaaring makipag-usap sa HTTP gamit ang dalawang makibang mga mode.
```
Comment (mind-blast) [inline book/04-git-server/sections/protocols.asc]: makibang  to magkaibang

File: book/04-git-server/sections/protocols.asc [hunk 6]
```diff
@@ -1,100 +1,100 @@
...
-Basically, all you have to do is put a bare Git repository under your HTTP document root and set up a specific `post-update` hook, and you're done (See <<_customizing_git#_git_hooks>>).
-At that point, anyone who can access the web server under which you put the repository can also clone your repository.
-To allow read access to your repository over HTTP, do something like this:
+(((protokol, dumb HTTP)))
+Kapag ang server ay hindi tumugon ng isang matalinong serbisyo ng Git HTTP, Susubukan ng kliyente ng Git na bumalik sa mas madaling _Dumb_ HTTP na protokol.
+Inaasahan ng Dumb na protokol ang hubad na repositoryo ng Git ang ihahatid tulad ng mga normal na mga file mula sa web server.
+Ang kagandahan ng Dumb HTTP ay ang madaling pag set-up nito.
+Sa totoo lang, ang kailangan mo lang gawin ay maglagay ng hubad na repositoryo ng Git sa ilalim ng isang HTTP na dokumento sa root at mag-set up ng isang tiyak na `post-update` hook, at tapos ka na (Tingnan <<_customizing_git#_git_hooks>>).
+Sa puntong iyon, sinuman na makaka-access sa web server sa ilalim na kung saan maaaring ilagay ang repositoryo ay maaaring i-clong ang iyong repositoryo.
```
Comment (mind-blast) [inline book/04-git-server/sections/protocols.asc]: i-clong to i-clone

File: book/04-git-server/sections/protocols.asc [hunk 7]
```diff
@@ -1,100 +1,100 @@
...
+Upang ma-allow ang access sa pagbasa sa iyong repositoryo sa HTTP, gumawa ng katulad nito:
```
Comment (mind-blast) [inline book/04-git-server/sections/protocols.asc]: ma-allow to payagan

File: book/04-git-server/sections/protocols.asc [hunk 8]
```diff
@@ -103,20 +103,20 @@ $ mv hooks/post-update.sample hooks/post-update
...
 
-That's all.(((hooks, post-update)))
-The `post-update` hook that comes with Git by default runs the appropriate command (`git update-server-info`) to make HTTP fetching and cloning work properly.
-This command is run when you push to this repository (over SSH perhaps); then, other people can clone via something like
+Yun Lang.(((mga hook, post-update)))
+Ang `post-update` na hook na kasama ng Git bilang default ay ipinapagana ang tamang utos (`git update-server-info`) upang mapagana ng maayos ang HTTP fetching ang pag-kopya.
+Ang utos na ito ay pinapagana kapag nag-push ka sa repositoryo na ito (marahil sa SSH); pagktapos, maaaring mag-clone ang ibang tao gamit ang katulad
 
-[source,console]
+[pinagmulan,console]
```
Comment (mind-blast) [inline book/04-git-server/sections/protocols.asc]: [source,console]

File: book/04-git-server/sections/protocols.asc [hunk 9]
```diff
@@ -1,100 +1,100 @@
...
 ===== Dumb HTTP
 
-(((protocols, dumb HTTP)))
-If the server does not respond with a Git HTTP smart service, the Git client will try to fall back to the simpler _Dumb_ HTTP protocol.
-The Dumb protocol expects the bare Git repository to be served like normal files from the web server.
-The beauty of Dumb HTTP is the simplicity of setting it up.
-Basically, all you have to do is put a bare Git repository under your HTTP document root and set up a specific `post-update` hook, and you're done (See <<_customizing_git#_git_hooks>>).
-At that point, anyone who can access the web server under which you put the repository can also clone your repository.
-To allow read access to your repository over HTTP, do something like this:
+(((protokol, dumb HTTP)))
```
Comment (mind-blast) [inline book/04-git-server/sections/protocols.asc]: (((protocols, dumb HTTP)))

File: book/04-git-server/sections/protocols.asc [hunk 10]
```diff
@@ -1,100 +1,100 @@
...
+Sa bersyon 1.6.6, isang bago, mas matalinong protokol ang ipinakilala na kasangkot ang Git na magagawang mas matalino na pag-usap ng pag lipat ng data sa parehong paraan ng SSH. 
+Sa mga nakaraang mga taon, itong bago na protokol ng HTTP ay naging popular dahil ito ay mas madali para sa user at mas matalino kung papaano nakikipag-usap.
+Ang mas bagong bersyon ay kadalasan mas tinutukoy bilang ang _Smart_ HTTP na protokol at ang lumang paraan ay ang _Dumb_ HTTP.
+Tatalakayin natin una ang mas bago na Smart HTTP protokol.
 
 ===== Smart HTTP
 
-(((protocols, smart HTTP)))
-Smart HTTP operates very similarly to the SSH or Git protocols but runs over standard HTTPS ports and can use various HTTP authentication mechanisms, meaning it's often easier on the user than something like SSH, since you can use things like username/password authentication rather than having to set up SSH keys.
+(((protokol, smart HTTP)))
```
Comment (mind-blast) [inline book/04-git-server/sections/protocols.asc]: (((protocols, smart HTTP)))

File: book/04-git-server/sections/protocols.asc [hunk 11]
```diff
@@ -1,100 +1,100 @@
-=== The Protocols
+=== Ang Mga Protokol
+Maaaring gamitin ng Git ang apat na magkakaibang mga protokol upang lumipat ng mga data: Lokal, HTTP, Secure Shell (SSH) at Git.
```
Comment (heywens) [inline book/04-git-server/sections/protocols.asc]: Add line break after ``=== Ang Mga Protokol``

File: book/04-git-server/sections/protocols.asc [hunk 12]
```diff
@@ -1,100 +1,100 @@
...
+Dito tatalakayin natin kung ano sila at kung saang mga pangunahing pangyayari mo gusto (o di gusto) sila gamitin.
 
-Git can use four distinct protocols to transfer data: Local, HTTP, Secure Shell (SSH) and Git.
-Here we'll discuss what they are and in what basic circumstances you would want (or not want) to use them.
+==== Lokal na Protokol
 
-==== Local Protocol
+(((mga protokol, lokal)))
+Ang pinaka-pangunahing ay ang _Lokal na protokol_, na kung saan ang remote na repositoryo ay nasa ibang direktoryo sa parehong host.
```
Comment (heywens) [inline book/04-git-server/sections/protocols.asc]: ``pinaka-pangunahing`` to ``pinaka-pangunahin``

File: book/04-git-server/sections/protocols.asc [hunk 13]
```diff
@@ -1,100 +1,100 @@
...
+Madalas itong ginagamit kapag lahat kayo sa inyong koponan ay may-access sa isang ibinahagi na filesystem tulad ng isang NFS mount, o sa malabong kaso na lahat ay nag-log sa parehong kompyuter.
```
Comment (heywens) [inline book/04-git-server/sections/protocols.asc]: ``may-access` to ``may access``

File: book/04-git-server/sections/protocols.asc [hunk 14]
```diff
@@ -1,100 +1,100 @@
...
+Ang huli ay hindi ideal, kasi lahat ng instansya ng repositoryo ng iyong code ay naninirahan sa parehong kompyuter, mas malaki ang posibilidad ng pagkawala ng sakuna.
```
Comment (heywens) [inline book/04-git-server/sections/protocols.asc]: ``ideal`` to ``uliran`` or ``tamang-tama``? not sure

File: book/04-git-server/sections/protocols.asc [hunk 15]
```diff
@@ -1,100 +1,100 @@
...
 
-(((protocols, local)))
-The most basic is the _Local protocol_, in which the remote repository is in another directory on the same host.
-This is often used if everyone on your team has access to a shared filesystem such as an NFS mount, or in the less likely case that everyone logs in to the same computer.
-The latter wouldn't be ideal, because all your code repository instances would reside on the same computer, making a catastrophic loss much more likely.
+Kapag ikaw ay may isang ibinahagi na naka-mount na filesystem, maaari kang mag clone, i-push sa, at mag pull mula sa isang lokal na nakabatay sa file na repositoryo.
```
Comment (heywens) [inline book/04-git-server/sections/protocols.asc]: > i-push sa, at mag pull
to
> mag-push at mag-pull

File: book/04-git-server/sections/protocols.asc [hunk 16]
```diff
@@ -1,100 +1,100 @@
...
 $ git clone file:///srv/git/project.git
 ----
 
-Git operates slightly differently if you explicitly specify `file://` at the beginning of the URL.
-If you just specify the path, Git tries to use hardlinks or directly copy the files it needs.
-If you specify `file://`, Git fires up the processes that it normally uses to transfer data over a network, which is generally much less efficient.
-The main reason to specify the `file://` prefix is if you want a clean copy of the repository with extraneous references or objects left out -- generally after an import from another VCS or something similar (see <<_git_internals#_git_internals>> for maintenance tasks).
-We'll use the normal path here because doing so is almost always faster.
+Ang Git ay gumagana ng bahagyang maiba kapag tahasang tinukoy mo ang `file://` sa simula ng URL.
+Kapag ang landas lamang ang tinukoy mo, Sinusubukan ng Git na gumamit ng mga hardlink o direktang kinokopya ang mga file na kailangan nito.
```
Comment (heywens) [inline book/04-git-server/sections/protocols.asc:30]: ``Sinusubukan`` to lower case

File: book/04-git-server/sections/protocols.asc [hunk 17]
```diff
@@ -1,100 +1,100 @@
...
+Kapag tinukoy mo ang `file://`, Pinapagana ng Git ang mga proseso na karanawing ginagamit upang lumipat ng data sa isang network, na kung saan ay karaniwang mas mahina.
```
Comment (heywens) [inline book/04-git-server/sections/protocols.asc]: ``Pinapagana`` > lowercase

File: book/04-git-server/sections/protocols.asc [hunk 18]
```diff
@@ -1,100 +1,100 @@
...
+Ang pinakarason upang tukuyin ang `file://` na prefix ay kung gusto mo ng isang malinis na kopya ng repositoryo na may mga reperensiya o mga object na naiwan -- Sa pangkalahatan ay pagkatapos ng pag-import mulang sa ibang VCS o sa isang bagay na katulad (tingnan <<_git_internals#_git_internals>> para sa mga gawain ng pagpapanatili).
```
Comment (heywens) [inline book/04-git-server/sections/protocols.asc]: ``mulang`` to ``mula``

File: book/04-git-server/sections/protocols.asc [hunk 19]
```diff
@@ -1,100 +1,100 @@
...
-A local repository is fast only if you have fast access to the data.
-A repository on NFS is often slower than the repository over SSH on the same server, allowing Git to run off local disks on each system.
+Mahalagang banggitin na hindi ito ang pinakamabilis na opsyon kung ikaw ay gumagamit ng isang uri ng ibinahagi na mount.
+Ang isang lokal na repositoryo ay mabilis lamang kapag ikaw ay may mabilis na access sa data.
+Ang repositoryo na nasa NFS ay madalas mas mahina kumpara sa repositoryo na nasa SSH sa parehong server, nagpapahintulot na ipagana ang Git gamit ang lokal na mga disk sa bawat sistema.
 
-Finally, this protocol does not protect the repository against accidental damage.
-Every user has full shell access to the ``remote'' directory, and there is nothing preventing them from changing or removing internal Git files and corrupting the repository.
+Sa wakas, hindi pinoprotektahan ng protokol na ito ang repositoryo mula sa mga hindi sinasadya na mga aksidente.
+Ang bawa't gumagamit ay may buong access ng shell sa ``remote'' na direktoryo, at walang pumipigil sa kanila sa pag bago o pag tanggal ng mga Git files sa loob at pag-corrupt sa repositoryo.
```
Comment (heywens) [inline book/04-git-server/sections/protocols.asc]: * ``bawa't`` to ``bawat``
* ``pag bago`` to ``pagbago``
* ``pag tanggal`` to ``pagtanggal``

File: book/04-git-server/sections/protocols.asc [hunk 20]
```diff
@@ -1,100 +1,100 @@
...
 
-Git can communicate over HTTP using two different modes.
-Prior to Git 1.6.6, there was only one way it could do this which was very simple and generally read-only.
-In version 1.6.6, a new, smarter protocol was introduced that involved Git being able to intelligently negotiate data transfer in a manner similar to how it does over SSH.
-In the last few years, this new HTTP protocol has become very popular since it's simpler for the user and smarter about how it communicates.
-The newer version is often referred to as the _Smart_ HTTP protocol and the older way as _Dumb_ HTTP.
-We'll cover the newer Smart HTTP protocol first.
+Ang Git ay maaaring makipag-usap sa HTTP gamit ang dalawang makibang mga mode.
+Bago ang Git 1.6.6, Mayroon lamang isang paraan upang maisagawa ito na kung saan ay madali at sa pangkalahatan ay read-only.
+Sa bersyon 1.6.6, isang bago, mas matalinong protokol ang ipinakilala na kasangkot ang Git na magagawang mas matalino na pag-usap ng pag lipat ng data sa parehong paraan ng SSH.
```
Comment (heywens) [inline book/04-git-server/sections/protocols.asc]: ``pag lipat`` to ``paglipat``

File: book/04-git-server/sections/protocols.asc [hunk 21]
```diff
@@ -1,100 +1,100 @@
...
-In the last few years, this new HTTP protocol has become very popular since it's simpler for the user and smarter about how it communicates.
-The newer version is often referred to as the _Smart_ HTTP protocol and the older way as _Dumb_ HTTP.
-We'll cover the newer Smart HTTP protocol first.
+Ang Git ay maaaring makipag-usap sa HTTP gamit ang dalawang makibang mga mode.
+Bago ang Git 1.6.6, Mayroon lamang isang paraan upang maisagawa ito na kung saan ay madali at sa pangkalahatan ay read-only.
+Sa bersyon 1.6.6, isang bago, mas matalinong protokol ang ipinakilala na kasangkot ang Git na magagawang mas matalino na pag-usap ng pag lipat ng data sa parehong paraan ng SSH. 
+Sa mga nakaraang mga taon, itong bago na protokol ng HTTP ay naging popular dahil ito ay mas madali para sa user at mas matalino kung papaano nakikipag-usap.
+Ang mas bagong bersyon ay kadalasan mas tinutukoy bilang ang _Smart_ HTTP na protokol at ang lumang paraan ay ang _Dumb_ HTTP.
+Tatalakayin natin una ang mas bago na Smart HTTP protokol.
```
Comment (heywens) [inline book/04-git-server/sections/protocols.asc]: > Tatalakayin natin una
to
> Una nating tatalakayin

**Review by heywens — COMMENTED** [review]
Hi @jjrang11, 

I've suggested few changes for your translation. Kindly check. Some suggestion needs validation. Kindly ask the project owner, @severinolorillajr.

Thanks.

**Comment by jjrang11** [comment]
Hey @robin-ho and @heywens, thank you so much for your inputs! I have since updated the file based on your suggestions

**Pull request merged**